### ISSUE TITLE
TypeError

### ISSUE THREAD

**Issue by halitalf** [Drewsif/OpenVPN-Config-Generator#1]
I was trying to generate configs for an OpenVPN server...
Version Info:
`Python 2.7.5`
```
OpenVPN 2.3.11 x86_64-redhat-linux-gnu [SSL (OpenSSL)] [LZO] [EPOLL] [PKCS11] [MH] [IPv6] built on May 10 2016
library versions: OpenSSL 1.0.1e-fips 11 Feb 2013, LZO 2.06
Originally developed by James Yonan
Copyright (C) 2002-2010 OpenVPN Technologies, Inc. <sales@openvpn.net>
Compile time defines: enable_crypto=yes enable_crypto_ofb_cfb=yes enable_debug=yes enable_def_auth=yes enable_dependency_tracking=no enable_dlopen=unknown enable_dlopen_self=unknown enable_dlopen_self_static=unknown enable_fast_install=yes enable_fragment=yes enable_http_proxy=yes enable_iproute2=yes enable_libtool_lock=yes enable_lzo=yes enable_lzo_stub=no enable_management=yes enable_multi=yes enable_multihome=yes enable_pam_dlopen=no enable_password_save=yes enable_pedantic=no enable_pf=yes enable_pkcs11=yes enable_plugin_auth_pam=yes enable_plugin_down_root=yes enable_plugins=yes enable_port_share=yes enable_pthread=yes enable_selinux=no enable_server=yes enable_shared=yes enable_shared_with_static_runtimes=no enable_small=no enable_socks=yes enable_ssl=yes enable_static=yes enable_strict=no enable_strict_options=no enable_systemd=yes enable_win32_dll=yes enable_x509_alt_username=yes with_aix_soname=aix with_crypto_library=openssl with_gnu_ld=yes with_iproute_path=/sbin/ip with_mem_check=no with_plugindir='$(libdir)/openvpn/plugins' with_sysroot=no
```
Error:
```
Which template would you like to use?
1) Basic: A basic setup, good for internet browsing
2) LAN Games: A basic setup for playing LAN games
Enter selection: 1
Enter a name for the configs: server
Enter Hostname/IP: removed.forprivacy.com
Enter port number: 443
Traceback (most recent call last):
  File "./OpenVPNConfig.py", line 351, in <module>
    main()
  File "./OpenVPNConfig.py", line 348, in main
    create_confs(name, template)
  File "./OpenVPNConfig.py", line 220, in create_confs
    cacert, cakey = create_ca()
  File "./OpenVPNConfig.py", line 45, in create_ca
    ca.sign(key, "sha256")
  File "/usr/lib/python2.7/site-packages/OpenSSL/crypto.py", line 1065, in sign
    evp_md = _lib.EVP_get_digestbyname(_byte_string(digest))
TypeError: initializer for ctype 'char *' must be a str or list or tuple, not unicode
```

**Comment by Drewsif** [comment]
Thanks for letting me know about the issue! Hopefully it should be fixed now. Please let me know if you run into any more issues :)

**Comment by halitalf** [comment]
Not a problem. I noticed that it had been over a year since the last commit, so I actually forked and proceeded to fix and improve. I don't have a whole lot of Python skill, but I think the end result looks good IMO. If you're interested in my modifications, https://github.com/halitalf/DynaGen

**Comment by Drewsif** [comment]
I like the idea of making it more automatable by allowing the templates to have certain options. I may make it done via command line arguments instead though. I'm also planning on adding the ability to change key sizes since I see you changed them around.

I do dislike the way I handle saving keys but I've not put enough thought into the best way to fix it. I use embedded keys exclusively and almost always do one client config to one server config.

Thanks for the ideas!

**Comment by halitalf** [comment]
I happen to like large keys, so I bumped it up. I will likely add the key size to the meta on my fork. I was using the script to generate the initial config as well as additional users. I will also be creating an extra branch at some point with a PHP/Cron creation/deletion queue. I essentially want to turn what you started into a full OpenVPN management suite. Anyway...
You're most definitely welcome for the ideas. I've enjoyed working with your code and talking with you. It was an excellent learning experience for Python.
